rsmtool/utils.py

```diff
@@ -586,6 +595,56 @@ def compute_subgroup_plot_params(group_names, num_plots):
     return (figure_width, figure_height, num_rows, num_columns, wrapped_group_names)
 
 
+def get_output_directory_extension(directory, experiment_id):
+    """
+    Check the output directory to determine what file extensions
+    exist. If more than one extension (in the possible list of
+    extensions) exists, then raise a ValueError. Otherwise,
+    return the one file extension. If no extensions can be found, then
+    `csv` will be returned by default.
+
+    Possible extensions include: `csv`, `tsv`, `xlsx`. Files in the
+    directory with none of these extensions will be ignored.
+
+    Parameters
+    ----------
+    directory : str
+        The path to the directory where output is located.
+    experiment_id : str
+        The ID of the experiment.
+
+    Returns
+    -------
+    extension : {'csv', 'tsv', 'xlsx'}
+        The extension that output files in this directory
+        end with.
+
+    Raises
+    ------
+    ValueError
+        If any files in the directory have different extensions,
+        and are in the list of possible output extensions.
+    """
+    extension = 'csv'
+    extensions_identified = [os.path.splitext(name)[-1].replace('.', '')
+                             for name in os.listdir(directory)
+                             if any(name.endswith(ext) for ext in POSSIBLE_EXTENSIONS)]
```

Comment:
This seems a little complicated and inefficient? How about a different approach:

```python
def has_files_with_extension(dir, ext):
    files_with_extension = glob(join(dir, f'*.{ext}'))
    return len(files_with_extension) > 0

extensions_identified = [ext for ext in extensions if has_files_with_extension(directory, ext)]
```